charts/datadog/templates/agent-services.yaml

```diff
@@ -102,7 +102,7 @@ spec:
  {{- end }}
 {{- if eq (include "should-enable-otel-agent" .) "true" }}
 {{- range .Values.datadog.otelCollector.ports }}
-    - protocol: TCP
+    - protocol: {{ .protocol | default "TCP" }}
```

Comment:
nit - default seems redundant since it's set to the `TCP` in values.yaml anyway.